src/modules/Packages.rb

```diff
@@ -788,30 +788,29 @@ def ForceFullRepropose
       nil
     end
 
-    # Reset package selection, but keep objects of specified type
-    # @param [Array<Symbol>] keep a list of symbols specifying type of objects to be kept
+    # Reset package selection, but keep the selected objects of the specified type
+    # @param [Array<Symbol>] keep a list of symbols specifying type of objects to be kept selected
     def Reset(keep)
-      keep = deep_copy(keep)
       restore = []
-      Builtins.foreach(keep) do |type|
-        selected = Pkg.ResolvableProperties("", type, "")
-        Builtins.foreach(selected) do |s|
-          restore = Builtins.add(
-            restore,
-            { "type" => type, "name" => Ops.get_string(s, "name", "") }
-          )
+
+      # collect the currently selected resolvables
+      keep.each do |type|
```

Comment:
But we need to run this loop for each `keep` value, we would need to merge the results....

```ruby
restore.merge(resolvables.each_object ....)
```

I'm not sure if that would better readable than the current code... :worried: